.github/workflows/build-and-deploy-container-image.yaml

```diff
@@ -476,6 +476,49 @@ jobs:
           gh-app-id: ${{ secrets.gh-app-id }}
           gh-app-priv-key: ${{ secrets.gh-app-private-key }}
 
+  status-badge:
+    if: "!contains(github.event.pull_request.body, 'App Deployment') && contains(github.event.pull_request.labels.*.name, 'deploy')"
+    runs-on: ubuntu-latest
+    needs:
+      - build-docker-image
+    timeout-minutes: 5
+    steps:
+      - name: Prepare
+        id: prepare
+        run: |
+          for element in $(echo ${{ github.event.repository.name }} | tr "-" "\n" | tr "[:upper:]" "[:lower:]"); do
+            abbreviation=$abbreviation"${element:0:1}"
+          done
+
+          echo ::set-output name=argocd-suffix::"-$abbreviation-${{ github.event.number }}"
+      - name: Comment on PR
+        uses: actions/github-script@v5
+        env:
+          PR_ADDITION: |
+            :tada: This PR is an App Deployment :tada: 
+            [![App Status](${{ inputs.argo-host }}/api/badge?name=${{ inputs.argo-base-app }}${{ steps.prepare.outputs.argocd-suffix }}&revision=true)](${{ inputs.argo-host }}/applications/${{ inputs.argo-base-app }}${{ steps.prepare.outputs.argocd-suffix }})
+            View the app at: https://server${{ steps.prepare.outputs.argocd-suffix }}.${{ inputs.workspaces-base-domain }}
+
+            ---
+            Please do not edit this part above
+        with:
+          script: |
+            const pr = await github.rest.pulls.get({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              pull_number: context.issue.number
+            });
+
+            const prBody = pr.data.body;
+            const newBody = process.env.PR_ADDITION + '\n' + prBody;
+            github.rest.pulls.update({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              pull_number: context.payload.number,
+              body: newBody
+            });
+          github-token: ${{ secrets.github-token }}
+
```

Comment:
Why don't we add this to a composite, so that we can re-use it here and for the PR deployments within gitops?

Additionally I would remove the status check `ArgoCD PR-Deployment`.